src/platform/platform_linux.cpp

```diff
@@ -166,3 +181,204 @@ bool mp::platform::is_image_url_supported()
 {
     return true;
 }
+
+mp::IPAddress get_ip_address(const std::string& iface_name)
+{
+    const char* iface = iface_name.c_str();
+    struct ifreq ifr;
+
+    int fd = socket(AF_INET, SOCK_DGRAM, 0);
+
+    ifr.ifr_addr.sa_family = AF_INET;
+    strncpy(ifr.ifr_name, iface, IFNAMSIZ - 1);
```

Comment:
IIUC, if the size is `IFNAMSIZ` the trailing `\0` will not be copied, but `ioctl` won't need that `\0` because it will read at most `IFNAMSIZ - 1` bytes.

You are proposing a cleaner C++ solution, and it makes sense to me. But I'm not sure the `strncpy` you suggest will work, because we need to copy the trailing `\0`, so the number of copied bytes should be `iface_name.size() + 1` if the size is smaller than `IFNAMSIZ`; `iface_name.size()` otherwise.

I implemented the throwing line only.

Reply:
> `ioctl` won't need that `\0` because it will read at most `IFNAMSIZ - 1` bytes.

Hmm, where does it say that it will read at most `IFNAMSIZE -1`?

> so the number of copied bytes should be `iface_name.size() + 1` if the size is smaller than `IFNAMSIZ`; `iface_name.size()` otherwise

Right, except that the `if` throws unless `iface_name.size() < IFNAMSIZ`, so we know we have space for the `\0`. Which we should copy, so there's a bug in my suggestion: it should be `strncpy(ifr.ifr_name, iface, iface_name.size() + 1);`

Alternativelly (or additionally, since the safety is worth paying for one init here), the struct could be zero-initialized by just adding braces: `struct ifreq ifr{};`.

Anyway, this is now outside of the scope of this PR, so this is just accademic for now.